cnd/src/main.rs

```diff
@@ -37,6 +38,11 @@ fn main() -> anyhow::Result<()> {
 
     let settings = read_config(&options).and_then(Settings::from_config_file_and_defaults)?;
 
+    if options.dump_config {
```

Comment:
What is a bit annoying about the rest of the code is that it also prints to `stdout`. This means redirecting the output with this option will _not_ yield a valid config file. Should we just change all the remaining print statements to `stderr`, including the logger?

Reply:
This can actually be done as a follow-up PR. However, if you do it now, we can write a nice test that simply does:

```bash
cargo run -p cnd -- --dump-config > ./my_config.toml
cargo run -p cnd -- --config ./my_config.toml
```

That verifies that all of this works :)

Reply:
I don't think it is an issue to have this going to stdout, in my opinion the use case for this is exploratory not to output a valid config file that can be re-parsed.  Either way there is no output other than valid config file if the `--config` option is not present.

In regard to the second post, I like a unit test that has the hard coded config file as as string better than one that writes to the filesystem, although I would not NACK a test that did this.